src/installer/protocols.py

```diff
@@ -0,0 +1,118 @@
+import abc
+from enum import Enum
+
+from installer._compat import with_metaclass
+from installer._compat.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from io import BufferedReader
+    from typing import List, Sequence, Tuple
+
+    from installer._compat.typing import Text, FSPath
+
+
+Scheme = Enum("Scheme", ["purelib", "platlib", "headers", "scripts", "data"])
```

Comment:
Do we need this to be an `Enum`? The available scheme keys are [well-defined in `sysconfig`](https://docs.python.org/3/library/sysconfig.html#installation-paths), and I imagine users are going to use `sysconfig.get_paths()` in the vast majorities of cases anyway. I feel this is not enough reason to pull in an extra dependency for Python 2. If type safty is the concern, `typing.Literal` would work well enough IMO.

Reply:
We don't. I'm 100% OK to change this. :)